sky/backends/backend_utils.py

```diff
@@ -1617,17 +1617,10 @@ def get_head_ssh_port(
 ) -> int:
     """Returns the ip of the head node."""
     del max_attempts  # Unused.
-    # Use port 22 for everything except Kubernetes
+    # Use port 22 for everything including Kubernetes.
+    # Note: for Kubernetes we use ssh jump host to access ray head
     # TODO(romilb): Add a get port method to the cloud classes.
     head_ssh_port = 22
-    if not isinstance(handle.launched_resources.cloud, clouds.Kubernetes):
-        return head_ssh_port
-    else:
-        if use_cache and handle.head_ssh_port is not None:
-            head_ssh_port = handle.head_ssh_port
-        else:
-            svc_name = f'{handle.get_cluster_name()}-ray-head-ssh'
-            head_ssh_port = clouds.Kubernetes.get_port(svc_name)
```

Comment:
Since, we are not using `handle` and `use_cache` now, shoud we remove these from `get_head_ssh_port`'s function signature, and other places where it passes `handle` and `use_cache` as an argument to the function call?

Reply:
@landscapepainter,
I looked into the sources and could see it is being referenced by several locations in `cloud_vm_ray_backend`. 
I think I would suggest to keep it as is in this PR and to gradually remove `handle` and `use_cache` later on.

Reply:
@aviweit If we decide to keep the parameters, we might as well need to keep the rest of the function. Unused parameters won't make it through the linting. 

I think we can keep your change and make the adjustments on where the function is referenced from the other locations, but I'll let @romilbhardwaj confirm on this.